controllers/usersignup/masteruserrecord.go

```diff
@@ -22,7 +22,7 @@ func migrateOrFixMurIfNecessary(mur *toolchainv1alpha1.MasterUserRecord, nstempl
 			changed = true
 		}
 		nsTemplateSet := userAccount.Spec.NSTemplateSet
-		if nsTemplateSet.TierName == "" {
+		if nsTemplateSet == nil || nsTemplateSet.TierName == "" {
```

Comment:
if the `nsTemplateSet` is nil, then it should not add the default `NSTemplateSet` - we want to keep it empty so `UserAccount` controller skips it, does it make sense?

Reply:
done in 34c4a4f